spec/helpers/ajax-helper.ts

```diff
@@ -218,4 +226,33 @@ export class MockXMLHttpRequest {
       }
     });
   }
-}
\ No newline at end of file
+}
+
+export class MockXMLHttpRequestInternetExplorer extends MockXMLHttpRequest {
+  constructor() {
+    super();
+  }
+
+  private mockHttp204() {
+    this.responseType = '';
+    this.responseText = '';
+    this.response = '';
+  }
+
+  protected jsonResponseValue(response: any) {
+    if (this.status == 204) {
+      this.mockHttp204();
+      return;
+    }
+    return super.jsonResponseValue(response);
+  }
+
+  protected defaultResponseValue() {
```

Comment:
There is a scenario where one can write a unit test like this:
```
request.respondWith({
  'status': 204,
  'contentType': 'application/json',
  'responseType': 'json',
  'responseText': expected
});
```

The issue is that the IE does not actually provide ```  'responseType': 'json', ``` so this would be a presumptuous test. That is why when we mock IE that we actually unset ```responseType``` in ```private mockHttp204()```

That is the reason the MockXMLHttpRequestInternetExplorer overrides both, just in case.